File: src/scitrader.marketdataserver/src/main/java/com/scitrader/marketdataserver/common/Utility/Guard.java
package com.scitrader.marketdataserver.common.Utility;

import com.scitrader.marketdataserver.common.ArgumentException;
import com.scitrader.marketdataserver.common.ArgumentNullException;

public class Guard {

  public static void assertNotNull(Object o, String message){
    if (o == null){
      throw new ArgumentNullException(message);
    }
  }

  public static void assertIsTrue(boolean expression, String message) {
    if (expression){
      throw new ArgumentException(message);
    }
  }
}

File: Kotlin/hello-kotlin-world/src/main/java/com/scitrader/marketdataserver/common/utility/Guard.kt
package com.scitrader.marketdataserver.common.utility

import com.scitrader.marketdataserver.common.ArgumentException
import com.scitrader.marketdataserver.common.ArgumentNullException

object Guard {

    fun assertNotNull(o: Any?, message: String) {
        if (o == null) {
            throw ArgumentNullException(message)
        }
    }

    fun assertIsTrue(expression: Boolean, message: String) {
        if (!expression) {
            throw ArgumentException(message)
        }
    }
}